cmd/konnectd/bootstrap.go

```diff
@@ -59,12 +58,34 @@ const (
 	apiTypeSignin  = "signin"
 )
 
-// bootstrap is a data structure to hold configuration required to start
-// konnectd.
-type bootstrap struct {
-	cmd  *cobra.Command
-	args []string
+//Stringly typed application config, represents the user accessible config params
+type Config struct {
+	ISS                            string
+	IdentityManager                string
+	URIBasePath                    string
+	SignInUri                      string
+	SignedOutUri                   string
+	AuthorizationEndpointURI       string
+	EndsessionEndpointURI          string
+	Insecure                       bool
+	TrustedProxy                   []string
+	AllowScope                     []string
+	AllowClientGuests              bool
+	AllowDynamicClientRegistration bool
+	EncryptionSecretFile           string
+	Listen                         string
+	IdentifierClientPath           string
+	IdentifierRegistrationConf     string
+	IdentifierScopesConf           string
+	SigningKid                     string
+	SigningMethod                  string
+	SigningPrivateKeyFiles         []string
+	ValidationKeysPath             string
+}
 
+// Bootstrap is a data structure to hold configuration required to start
+// konnectd.
+type Bootstrap struct {
```

Comment:
Not sure how i feel about people importing from "cmd/konnectd". Would you be comfortable with that? Guess it make sense and all but also means that the "cmd" suddenly becomes provided or a public interface. Originally i thought it might be best to make `bootstrap` package but i also like this to be connected to the actual command. Any more thoughts?

Reply:
Sorry this is supposed to be a draft PR, not sure how it got switched back to ready . I plan to do the config decoupling first and then extract the package. Also no linting is applied as of yet.